client/packages/programs/src/JsonForms/useJsonForms.tsx

```diff
@@ -140,6 +149,10 @@ export const useJsonForms = <R,>(
     try {
       const sanitizedData = stripEmptyAdditions(initialData, data);
       const result = await save?.(sanitizedData);
+      const actions = Object.values(submitActions.current);
+      for (const action of actions) {
+        await action();
+      }
```

Comment:
Seeing a possibility where we might want to be able to choose whether to run an action before or after the JSON form save, but I think this is fine for now. Probably wouldn't want to create the prescription if the JSON form submit gave an error.

Should be ok to create the prescription ID ahead of time to save on the JSON form (to look up prescription in future), then use that to create the prescription.

Reply:
Yep, currently they get registered with just a key and an action. But I was envisaging a number of other properties that (when we need it), we could also register:
- pre/post save
- priority (a number, where higher numbers are executed first, similar to the "testers" priority in JSON forms)

These could be passed through via options, so could be configurable in the JSON UI schema. Will have to wait and see whether this becomes necessary, but fairly easy to add if we need it! :)

Reply:
> Probably wouldn't want to create the prescription if the JSON form submit gave an error.

Similarly, we might need to save additional data to the JSON form on execution of the action, e.g. the ID of the prescription. I'll play around with this when doing the Prescription component.